compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -936,6 +946,19 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
         }
     }
 
+    /// Assume that the pointer points to the first chunk of a `TypeId` and return the type that its
+    /// provenance refers to, as well as the segment of the hash that this pointer covers.
```

Comment:
```suggestion
    /// Takes a pointer that is the first ptr-sized chunk of a `TypeId` and return the type that its
    /// provenance refers to, as well as the segment of the hash that this pointer covers.
```
The pointer does not *point to*` a `TypeId`, it is stored inside a `TypeId`.